src/metabase/async/streaming_response.clj

```diff
@@ -96,13 +92,13 @@
                (try
                  (do-f* f os finished-chan canceled-chan)
                  (catch Throwable e
-                   (log/error e "bound-fn caught unexpected Exception")
-                   (a/>!! finished-chan :unexpected-error))
+                   (log/error e "Caught unexpected Exception in streaming response body")
+                   (a/>!! finished-chan :unexpected-error)
+                   (write-error! os e nil))
                  (finally
                    (a/>!! finished-chan (if (a/poll! canceled-chan)
                                           :canceled
                                           :completed))
-                   (a/close! finished-chan)
```

Comment:
Since `finished-chan` is a promise-chan, it's not necessary to close it immediately after a value was put into it with `>!!`.

Reply:
True, but it seems like better practice to close all channels, even if we know it's a no-op sometimes. It feels like a latent bug if the kind of channel were different sometimes.

Reply:
To protect against accidental swap of `promise-chan` to `chan` I'd prefer an assertion rather than a just-in-case close!, but OK, this is not important, I'll remove the changes.

Reply:
Done.